packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputBoolean.tsx

```diff
@@ -21,6 +25,32 @@ export function GridFilterInputBoolean(props: GridFilterInputValueProps & TextFi
     setFilterValueState(item.value || '');
   }, [item.value]);
 
+  if (isSelectNative) {
+    return (
+      <rootProps.components.BaseTextField
+        label={apiRef.current.getLocaleText('filterPanelInputLabel')}
+        value={filterValueState}
+        onChange={onFilterChange}
+        variant="standard"
+        select
+        SelectProps={{
+          native: true,
+          ...rootProps.componentsProps?.baseSelect,
+        }}
```

Comment:
We could keep the same `rootProps.components.BaseTextField` but only have a `SelectProps` object which attributes vary depending on `isSelectNative`. Also, to avoid code duplication.